kernel/kernel-api/src/test/scala/io/delta/kernel/coordinatedcommits/InMemoryCommitCoordinatorClient.scala

```diff
@@ -0,0 +1,319 @@
+   * |----------------------------|------------------|---------------------------|
+   */
+  private class PerTableData(
+      var maxCommitVersion: Long = -1,
+      var active: Boolean = false) {
+    def updateLastRatifiedCommit(commitVersion: Long): Unit = {
+      this.active = true
+      this.maxCommitVersion = commitVersion
+    }
+
+    /**
+     * Returns the last ratified commit version for the table. If no commits have been done from
+     * commit-coordinator yet, returns -1.
+     */
+    def lastRatifiedCommitVersion: Long = if (!active) -1 else maxCommitVersion
+
+    // Map from version to Commit data
+    val commitsMap: TreeMap[Long, Commit] = new TreeMap[Long, Commit]
+    // We maintain maxCommitVersion explicitly since commitsMap might be empty
+    // if all commits for a table have been backfilled.
+    val lock: ReentrantReadWriteLock = new ReentrantReadWriteLock()
+  }
+
+  private val perTableMap = new ConcurrentHashMap[String, PerTableData]()
+
+  /////////////////
+  // Public APIs //
+  /////////////////
+
+  override def registerTable(
+      engine: Engine,
+      logPath: String,
+      tableIdentifier: Optional[TableIdentifier],
+      currentVersion: Long,
+      currentMetadata: AbstractMetadata,
+      currentProtocol: AbstractProtocol): JMap[String, String] = {
+    requireNonNull(engine, "engine is null")
+    requireNonNull(logPath, "logPath is null")
+    requireNonNull(tableIdentifier, "tableIdentifier is null")
+    requireNonNull(currentMetadata, "currentMetadata is null")
+    requireNonNull(currentProtocol, "currentProtocol is null")
+
+    val newPerTableData = new PerTableData(currentVersion + 1)
+    perTableMap.compute(logPath, (_, existingData) => {
+      if (existingData != null) {
+        if (existingData.lastRatifiedCommitVersion != -1) {
+          throw new IllegalStateException(
+            s"Table $logPath already exists in the commit-coordinator.")
+        }
+        // If lastRatifiedCommitVersion is -1 i.e. the commit-coordinator has never
+        // attempted any commit for this table => this table was just pre-registered. If
+        // there is another pre-registration request for an older version, we reject it and
+        // table can't go backward.
+        if (currentVersion < existingData.maxCommitVersion) {
+          throw new IllegalStateException(
+            s"Table $logPath already registered with commit-coordinator")
+        }
+      }
+      newPerTableData
+    })
+    Collections.emptyMap[String, String]()
+  }
+
+  override def commit(
+      engine: Engine,
+      tableDescriptor: TableDescriptor,
+      commitVersion: Long,
+      actions: CloseableIterator[Row],
+      updatedActions: UpdatedActions): CommitResponse = {
+    requireNonNull(engine, "engine is null")
+    requireNonNull(tableDescriptor, "tableDescriptor is null")
+    requireNonNull(actions, "actions is null")
```

Comment:
Why skip the check for `commitVersion`?

Reply:
See my comment below. the Java interface is a `long` not a `Long` -- it cannot be null